vaults: Bug in vaultfs causing panics on 32-bit (unaligned 64-bit atomic operation)

This is part of the cause for https://github.com/hairyhenderson/gomplate/issues/2121 - looks like the `vaultfs.refCountedClient` struct has an unaligned field.

```
panic: unaligned 64-bit atomic operation [recovered]
	panic: unaligned 64-bit atomic operation
goroutine 112 [running]:
testing.tRunner.func1.2({0x854a240, 0x868cb58})
	/usr/local/go/src/testing/testing.go:1631 +0x2a3
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1634 +0x434
panic({0x854a240, 0x868cb58})
	/usr/local/go/src/runtime/panic.go:770 +0x103
runtime/internal/atomic.panicUnaligned()
	/usr/local/go/src/runtime/internal/atomic/unaligned.go:8 +0x2d
runtime/internal/atomic.Xadd64(0x9d882a4, 0x1)
	/usr/local/go/src/runtime/internal/atomic/atomic_386.s:125 +0x11
github.com/hairyhenderson/go-fsimpl/vaultfs.(*refCountedClient).AddRef(...)
	/__w/go-fsimpl/go-fsimpl/vaultfs/client.go:27
github.com/hairyhenderson/go-fsimpl/vaultfs.newVaultFile(...)
	/__w/go-fsimpl/go-fsimpl/vaultfs/vault.go:200
github.com/hairyhenderson/go-fsimpl/vaultfs.vaultFS.Open({{0x869154c, 0x9e2d770}, 0x9efab40, {0x868dcd4, 0x9d882c0}, 0x9d882a0}, {0x868a844, 0x1})
	/__w/go-fsimpl/go-fsimpl/vaultfs/vault.go:171 +0xb8
testing/fstest.(*fsTester).openDir(0x9e61d14, {0x868a844, 0x1})
	/usr/local/go/src/testing/fstest/testfs.go:115 +0x3a
testing/fstest.(*fsTester).checkDir(0x9e61d14, {0x868a844, 0x1})
	/usr/local/go/src/testing/fstest/testfs.go:134 +0xda
testing/fstest.testFS({0x868e554, 0xa09a420}, {0x9e61f04, 0x5, 0x5})
	/usr/local/go/src/testing/fstest/testfs.go:64 +0x62
testing/fstest.TestFS({0x868e554, 0xa09a420}, {0x9e61f04, 0x5, 0x5})
	/usr/local/go/src/testing/fstest/testfs.go:37 +0x4e
github.com/hairyhenderson/go-fsimpl/internal/tests/integration.TestVaultFS(0x9d6a408)
	/__w/go-fsimpl/go-fsimpl/internal/tests/integration/vaultfs_test.go:161 +0x8a3
testing.tRunner(0x9d6a408, 0x8601124)
	/usr/local/go/src/testing/testing.go:1689 +0x125
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:1742 +0x3b9
```

--- a/vaultfs/client.go
+++ b/vaultfs/client.go
@@ -18,13 +18,13 @@ func newRefCountedClient(c *api.Client) *refCountedClient {
 // other files are still open.
 type refCountedClient struct {
 	*api.Client
-	refs uint64
+	refs atomic.Uint64
 }
 
 // var _ VaultClient = (*refCountedClient)(nil)
 
 func (c *refCountedClient) AddRef() {
-	atomic.AddUint64(&c.refs, 1)
+	c.refs.Add(1)
 }
 
 func (c *refCountedClient) RemoveRef() {
@@ -36,9 +36,9 @@ func (c *refCountedClient) RemoveRef() {
 	}
 
 	// decrements the ref count by overflowing (see atomic.AddUint64 doc)
-	atomic.AddUint64(&c.refs, ^uint64(0))
+	c.refs.Add(^uint64(0))
 }
 
 func (c *refCountedClient) Refs() uint64 {
-	return atomic.LoadUint64(&c.refs)
+	return c.refs.Load()
 }

